src/test/java/org/mule/test/http/BasicHttpTestCase.java

```diff
@@ -160,7 +160,7 @@ public void invalidError() throws Exception {
     public Event process(Event event) throws MuleException {
       Message message = event.getMessage();
       Object payload = message.getPayload().getValue();
-      assertThat(payload, is(nullValue()));
+      assertThat(payload, is(notNullValue()));
```

Comment:
I've changed the behaviour of the servers to match the client, now an empty response will be an empty stream/byte array instead of null.